### ISSUE TITLE
Gogs chooses the wrong README sometimes

### ISSUE THREAD

**Issue by jiphex** [gogs/gogs#877]
If I have a repo with a file named (for example) `README-stuff.md` as well as `README.md`, Gogs will choose to display the former instead of the latter which would be expected. This is different from the behaviour of say, Github.

It looks like this is because the [check](https://github.com/gogits/gogs/blob/9a1fe801e526ae1bc0a211d8a218aabcf985376d/modules/base/markdown.go#L71) for whether a file is a README just checks if `filename[:6]` is "README", and the [view](https://github.com/gogits/gogs/blob/master/routers/repo/view.go#L98) gives up when it finds the first file.

I've used this for extra documentation about specific aspects of the project, e.g README-Building.md. Github does the right thing here.

**Comment by Unknwon** [comment]
Thanks your feedback!

You're right, Gogs should also check that the 7th character is `.` as my quick think of solution.

**Comment by jiphex** [comment]
Wouldn't that break repos which just have a file named "README"? (no .md)

**Comment by Unknwon** [comment]
Good catch, just check length first, then see if the 7th is `.` or not.

**Comment by Unknwon** [comment]
Hi @jiphex , @jcracknell , I've pushed fix on `dev` branch and feel free to test on https://try.gogs.io/

**Comment by jiphex** [comment]
Seems OK: https://try.gogs.io/testzz/issue877

**Comment by Unknwon** [comment]
Great!

**Comment by AdrianVerde** [comment]
It still chooses the wrong README as in the case of this mirror
https://try.gogs.io/AdrianVerde/PHP
Where it prefers "README.REDIST.BINS" instead of "README.md"
